### PR TITLE
Document Async Example DAG

### PR DESCRIPTION
**Pull request by pankajastro**
- Add custom extension to generate example DAG from integration test

Redshift operator Example Generated Page
Screenshot 2022-06-26 at 8 58 28 PM

closes: https://github.com/astronomer/astronomer-providers/issues/474

### CODE REVIEW

File: astronomer/providers/amazon/aws/example_dags/example_redshift_cluster_management.py [hunk 1]
```diff
@@ -191,14 +191,14 @@ def delete_redshift_cluster_snapshot_callable() -> None:
...
-    # [START howto_operator_redshift_cluster_sensor_async]
+    # [START howto_senor_redshift_cluster_async]
     redshift_sensor = RedshiftClusterSensorAsync(
         task_id="redshift_sensor",
         cluster_identifier=REDSHIFT_CLUSTER_IDENTIFIER,
         target_status="available",
         aws_conn_id=AWS_CONN_ID,
     )
-    # [END howto_operator_redshift_cluster_sensor_async]
+    # [END howto_senor_redshift_cluster_async]
```
Comment (bharanidharan14) [inline astronomer/providers/amazon/aws/example_dags/example_redshift_cluster_management.py:201]: `sensor` spelling, change  `howto_senor_redshift_cluster_async` to `howto_sensor_redshift_cluster_async`

File: astronomer/providers/amazon/aws/example_dags/example_redshift_cluster_management.py [hunk 2]
```diff
@@ -191,14 +191,14 @@ def delete_redshift_cluster_snapshot_callable() -> None:
     )
     # [END howto_operator_redshift_resume_cluster_async]
 
-    # [START howto_operator_redshift_cluster_sensor_async]
+    # [START howto_senor_redshift_cluster_async]
```
Comment (bharanidharan14) [inline astronomer/providers/amazon/aws/example_dags/example_redshift_cluster_management.py:194]: sensor spelling, change howto_senor_redshift_cluster_async to howto_sensor_redshift_cluster_async

File: docs/providers/amazon/aws/sensors/emr.rst [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+.. exampleinclude:: /../astronomer/providers/amazon/aws/example_dags/example_emr_eks_containers_job.py
+    :language: python
+    :dedent: 4
+    :start-after: [START howto_sensor_emr_job_container_async]
+    :end-before: [END howto_sensor_emr_job_container_async]
+
+
+.. _howto/sensor:EmrJobFlowSensorAsync:
+
+To waits asynchronously for job running on EMR container to reach the terminal state
```
Comment (bharanidharan14) [inline docs/providers/amazon/aws/sensors/emr.rst:18]: Polls asynchronously for job running on EMR container to reach the terminal state.

File: docs/providers/amazon/aws/sensors/emr.rst [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+.. exampleinclude:: /../astronomer/providers/amazon/aws/example_dags/example_emr_sensor.py
+    :language: python
+    :dedent: 4
+    :start-after: [START howto_sensor_emr_job_flow_async]
+    :end-before: [END howto_sensor_emr_job_flow_async]
+
+
+.. _howto/sensor:EmrStepSensorAsync:
+
+To waits asynchronously until state of the EMR cluster step reaches any of the target states
```
Comment (bharanidharan14) [inline docs/providers/amazon/aws/sensors/emr.rst:30]: same here

File: docs/providers/amazon/aws/sensors/redshift_cluster.rst [hunk 5]
```diff
@@ -0,0 +1,13 @@
+Redshift Sensor Async
+"""""""""""""""""""""
+
+.. _howto/sensor/RedshiftClusterSensorAsync:
+
+To waits asynchronously for Redshift cluster to reach a specific status
```
Comment (bharanidharan14) [inline docs/providers/amazon/aws/sensors/redshift_cluster.rst:6]: same here

File: docs/providers/amazon/aws/sensors/redshift_cluster.rst [hunk 6]
```diff
@@ -0,0 +1,13 @@
...
+:class:`~astronomer.providers.amazon.aws.sensor.redshift_cluster.RedshiftClusterSensorAsync`.
+
+.. exampleinclude:: /../astronomer/providers/amazon/aws/example_dags/example_redshift_cluster_management.py
+    :language: python
+    :dedent: 4
+    :start-after: [START howto_senor_redshift_cluster_async]
```
Comment (bharanidharan14) [inline docs/providers/amazon/aws/sensors/redshift_cluster.rst:12]: correct the spelling for `sensor`

File: docs/_ext/exampleinclude.py [hunk 7]
```diff
@@ -0,0 +1,47 @@
+from docutils import nodes
+from docutils.parsers.rst import directives
+from sphinx.directives.code import LiteralIncludeReader
+from sphinx.util.docutils import SphinxDirective
+from sphinx.util.nodes import set_source_info
+
+
+class ExampleInclude(SphinxDirective):
```
Comment (kaxil) [inline docs/_ext/exampleinclude.py:8]: What are the benefits we get with this custom extension as compared to the Sphinx `literalinclude` directive?

Comment (kaxil) [reply docs/_ext/exampleinclude.py:8]: Example:

```
.. literalinclude:: example.txt
   :start-after: # START
   :end-before: # END
```

File: docs/_ext/exampleinclude.py [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+    }
+
+    def run(self):
+        """Extract text and return list of node"""
+        location = self.state_machine.get_source_and_line(self.lineno)
+        rel_filename, filename = self.env.relfn2path(self.content[0])
+        reader = LiteralIncludeReader(filename, self.options, self.config)
+        text, lines = reader.read(location=location)
+        example_git_location = f"# {self.git_repo}{rel_filename[3:]}"
+        text = text.__add__(example_git_location)
```
Comment (pankajastro) [inline docs/_ext/exampleinclude.py:36]: This is currently coming as text but I want to make it a clickable link like HTML href tag.  
@kaxil any thought on it how I can achieve it?

Screenshot 2022-06-27 at 7 55 35 PM

Comment (kaxil) [reply docs/_ext/exampleinclude.py:36]: in Airflow we use https://www.sphinx-doc.org/en/master/usage/extensions/viewcode.html -> https://github.com/apache/airflow/blob/main/docs/exts/exampleinclude.py

File: docs/providers/amazon/aws/operators/batch.rst [hunk 9]
```diff
@@ -0,0 +1,13 @@
+Batch Operator Async
+"""""""""""""""""""""""""""""""""""
+
+.. _howto/sensor/BatchOperatorAsync:
+
+To execute a job asynchronously on AWS Batch
+:class:`~astronomer.providers.amazon.aws.operators.batch.BatchOperatorAsync`.
+
+.. literalinclude:: /../astronomer/providers/amazon/aws/example_dags/example_batch.py
```
[the remark this answers is not in the document]
Comment (pankajastro) [reply docs/providers/amazon/aws/operators/batch.rst]: Fixed

File: docs/_ext/exampleinclude.py (hunk 7 above)
Comment (pankajastro) [reply docs/_ext/exampleinclude.py:8]: I'll address it to make the link clickable in separate PR tracking it here https://github.com/astronomer/astronomer-providers/issues/484

**Review by kaxil — APPROVED** [review]

**Pull request merged**